src/util/router.js

```diff
@@ -79,6 +79,49 @@ export const forceReplace = ({ router, unsavedChanges }, location) => {
   return router.replace(location);
 };
 
+// Returns a scrollBehavior function to pass to createRouter().
+export const createScrollBehavior = (animation = 'smooth') => {
+  // Function to cancel a previous attempt to scroll
+  let cancel;
+  const scrollTo = (scrollId) => {
+    const el = document.querySelector(`[data-scroll-id="${scrollId}"]`);
+    return el != null ? { el, top: 10, behavior: animation } : null;
+  };
+  return (to, from, savedPosition) => {
+    if (cancel != null) cancel();
+    if (to.hash === '') return to.path === from.path ? savedPosition : null;
```

Comment:
I wasn't able to test `savedPosition` because we use memory history mode in history, and `savedPosition` is based on `popstate` navigation. However, I verified locally that `savedPosition` is used.